### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by sidforhbc**

### CODE REVIEW

File: src/reducers/UIStateReducers.js [hunk 1]
```diff
@@ -11,88 +8,18 @@ const initialState = {
...
-    return state;
-  }
-
-  return assign({}, state, { shouldDisplayWaitListConfirmation: false });
-}
 export default (state = initialState, action) => {
   switch (action.type) {
     case ProductDetailsConstants.ActionTypes.REQUEST_ADD_TO_BAG:
-      state = assign({}, state, { isAddingItemToCart: true });
+      state = Object.assign({}, state, { isAddingItemToCart: true });
```
Comment (hbcscott) [inline src/reducers/UIStateReducers.js]: This change shouldn't be an issue as they had to update AEM to contain Object.assign at one point due to server side errors.  I am assuming that fix is still there.

File: src/utilities/__tests__/ADAUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { KeyCodes } from '@hbc/hbc-browser-constants';
+import { emulateButtonKeyUp, emulateButtonKeyDown } from '../ADAUtils.js';
+
+describe('ADAUtils', () => {
```
Comment (hbcscott) [inline src/utilities/__tests__/ADAUtils.test.js]: Out of curiosity why use key up vs key down in some situations

File: src/reducers/__tests__/__snapshots__/HeaderReducers.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+`;
+
+exports[`HeaderReducers should handle TRIGGER_ADD_ACCOUNT_LEVEL 1`] = `
+Object {
+  "accountInfo": Object {},
+  "headerNavProps": Object {
+    "headerNavHamburgerProps": Object {
+      "firstLevel": Array [],
+      "secondLevel": Array [
+        undefined,
```
Comment (hbcscott) [inline src/reducers/__tests__/__snapshots__/HeaderReducers.test.js.snap]: I think this undefined may come in when a node is removed due to hide logic

**Comment by track0x1** [comment]
Need to use the package-lock in `master`. All deps should come from artifactory otherwise it will fail CI as right now:
> npm ERR! sha1-hBs708X6QN7a7YutxSaXu2CfP3k= integrity checksum failed when using sha1: wanted sha1-hBs708X6QN7a7YutxSaXu2CfP3k= but got sha1-DX+YEJ8vQXmDCxlNo2yWGK8xJvo=. (14894 bytes)

File: src/actions/__tests__/InputCompletionActionCreators.test.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  it('should create a RECEIVE_LOOKAHEAD_SEARCH when response does not return search options', () => {
+    mockAxios.onGet(`${payload.searchLookaheadUrl}?term=${payload.value}`).reply(200, null);
+
+    return store.dispatch(InputCompletionActionCreators.handleLookaheadSearch(payload)).then(() => {
+      const actions = store.getActions();
+      expect(actions).toMatchSnapshot();
+    });
+  });
+
+  it('should create a RECEIVE_LOOKAHEAD_SEARCH when response is not an object', () => {
```
[the remark this answers is not in the document]
Comment (sidforhbc) [reply src/actions/__tests__/InputCompletionActionCreators.test.js]: This is for when the search request returns a 200 but there are no results

File: src/reducers/UIStateReducers.js (hunk 1 above)
Comment (sidforhbc) [reply src/reducers/UIStateReducers.js]: ok, i'm going to say ignore AEM constraints for now but good to know!

File: src/utilities/__tests__/ADAUtils.test.js (hunk 2 above)
Comment (sidforhbc) [reply src/utilities/__tests__/ADAUtils.test.js]: ADA team will be redoing all this

**Pull request merged**